src/services/QueryService.js

```diff
@@ -55,130 +45,68 @@ export const queryHazardUnitTableAsync = async (units) => {
   return responseJson.features.map(feature => feature.attributes);
 };
 
-export const queryReferenceTableAsync = async (units) => {
-  console.log('QueryService.queryReferenceTableAsync');
-
-  const url = config.urls.hazardReferenceTextTable;
+export const queryHazardUnitTableAsync = async (units) => {
+  console.log('QueryService.queryHazardUnitTableAsync');
 
-  units = new Set(units.map(unit => getHazardCodeFromUnitCode(unit)));
+  const where = `HazardUnit IN ('${units.join('\',\'')}')`;
+  const outFields = 'HazardName,HazardUnit,HowToUse,Description';
 
-  const whereClause = `Hazard IN ('${Array.from(units).join('\',\'')}')`;
+  return await queryTable(config.urls.hazardUnitTextTable, where, outFields);
+};
 
-  const parameters = {
-    where: whereClause,
-    outFields: 'Hazard,Text',
-    returnGeometry: false,
-    returnCentroid: false,
-    f: 'json'
-  };
+export const queryReferenceTableAsync = async (units) => {
+  console.log('QueryService.queryReferenceTableAsync');
 
-  const response = await fetch(`${url}/query?${stringify(parameters)}`);
-  const responseJson = await response.json();
+  units = getDistinctHazardCodesFromUnits(units);
+  const where = `Hazard IN ('${Array.from(units).join('\',\'')}')`;
+  const outFields = 'Hazard,Text';
 
-  return responseJson.features.map(feature => feature.attributes);
+  return queryTable(config.urls.hazardReferenceTextTable, where, outFields);
```

Comment:
This async modifier can be removed or you'll need to await something in the method.

```suggestion
export const queryReferenceTableAsync = (units) => {
```

```suggestion
  return await queryTable(config.urls.hazardReferenceTextTable, where, outFields);
```

You are awaiting all of these in a promise in app.js so I think returning the promise is fine?